benchmarks/game_of_life_concurrent/procedural_c#/Program.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Threading;
+
+namespace procedural_c_
+{
+	class Program
+	{
+		static int runs = 100;
+		static int height = 256;
+		static int width = 256;
+		static int logicalProcessors = Environment.ProcessorCount;
+		static bool[,] board = new bool[width, height];
+
+		static void Main(string[] args)
+		{
+			initilizeBoard();
+			for (int i = 0; i < runs; i++)
+			{
+				updateBoard();
+			}
+			var count = countAlive();
+			Console.WriteLine("Alive: " + count);
+		}
+
+		public static int countLiveNeighbors(int x, int y)
+		{
+			int value = 0;
+			for (int j = -1; j <= 1; j++)
+			{
+				int k = (y + j + height) % height;
+				for (int i = -1; i <= 1; i++)
+				{
+					int h = (x + i + width) % width;
+					if (board[h, k])
+					{
+						value++;
+					}
+				}
+			}
+
+			if (board[x, y])
+				value--;
+			return value;
+		}
+
+		public static void updateBordPartly(int start, int stop, bool[,] newBoard)
+		{
+			for (int i = start; i <= stop; i++)
+			{
+				var x = (i / width);
+				var y = (i % width);
+				var n = countLiveNeighbors(x, y);
+				var c = board[x, y];
+				newBoard[x, y] = c && (n == 2 || n == 3) || (!c && n == 3);
+			}
+		}
+
+		public static void updateBoard()
+		{
+			var newBoard = new bool[width, height];
+
+			Thread[] threadPool = new Thread[logicalProcessors];
+
+			for (int i = 0; i < logicalProcessors; i++)
+			{
+				var start = i * (width * height) / logicalProcessors;
+				var stop = ((i + 1) * (width * height) / logicalProcessors) - 1;
+				if (i == logicalProcessors)
+				{
+					stop--;
+				}
+
+				var thread1 = new Thread(() => updateBordPartly(start, stop, newBoard));
+				threadPool[i] = thread1;
+				thread1.Start();
+			}
+
+			for (int i = 0; i < logicalProcessors; i++)
+			{
+				threadPool[i].Join();
+			}
+
+			board = newBoard;
+		}
```

Comment:
Burde man bruge Parallel.For til det her i stedet for? Det ville i hvert fald være simplere kode og jeg tænker ikke det er uden for det procedurale paradigme...

Reply:
Men det her er jo bare opsætningen af tråde, ikke udførslen. Så det skal ikke gøres parallelt